src/pg/brc20/brc20-pg-store.ts

```diff
@@ -273,6 +321,42 @@ export class Brc20PgStore extends BasePgStoreModule {
       );
   }
 
+  async rollBackInscription(args: { inscription: DbInscription }): Promise<void> {
+    const activities = await this.sql<DbBrc20Activity[]>`
+      SELECT * FROM brc20_events WHERE inscription_id = ${args.inscription.id}
+    `;
+    if (activities.count === 0) return;
+    // Traverse all activities generated by this inscription and roll back actions that are NOT
+    // otherwise handled by the ON DELETE CASCADE postgres constraint.
+    for (const activity of activities) {
+      switch (activity.operation) {
+        case 'mint':
+          await this.rollBackMint(activity);
+          break;
+      }
+    }
+  }
+
+  private async rollBackMint(activity: DbBrc20MintActivity): Promise<void> {
+    // Get real minted amount and substract from places.
+    await this.sql`
+      WITH minted_balance AS (
+        SELECT address, avail_balance
+        FROM brc20_balances
+        WHERE inscription_id = ${activity.inscription_id} AND type = ${DbBrc20BalanceTypeId.mint}
+      ),
+      token_balance_update AS (
+        UPDATE brc20_deploys
+        SET minted_supply = minted_supply - (SELECT avail_balance FROM minted_balance)
+        WHERE id = ${activity.brc20_deploy_id}
+      )
+      UPDATE brc20_total_balances SET
+        avail_balance = avail_balance - (SELECT avail_balance FROM minted_balance),
+        total_balance = total_balance - (SELECT avail_balance FROM minted_balance)
+      WHERE address = (SELECT address FROM minted_balance)
+    `;
+  }
```

Comment:
@janniks this is how I think we can deal with rollbacks... what do you think? feel free to branch off this branch to add your own logic. I'll be complimenting this PR